InvenTree/stock/serializers.py

```diff
@@ -641,6 +643,100 @@ def save(self):
         )
 
 
+class StockChangeStatusSerializer(serializers.Serializer):
+    """Serializer for changing status of multiple StockItem objects"""
+
+    class Meta:
+        """Metaclass options"""
+        fields = [
+            'items',
+            'status',
+            'note',
+        ]
+
+    items = serializers.PrimaryKeyRelatedField(
+        queryset=StockItem.objects.all(),
+        many=True,
+        required=True,
+        allow_null=False,
+        label=_('Stock Items'),
+        help_text=_('Select stock items to change status'),
+    )
+
+    def validate_items(self, items):
+        """Validate the selected stock items"""
+
+        if len(items) == 0:
+            raise ValidationError(_("No stock items selected"))
+
+        return items
+
+    status = serializers.ChoiceField(
+        choices=InvenTree.status_codes.StockStatus.items(),
+        default=InvenTree.status_codes.StockStatus.OK.value,
+        label=_('Status'),
+    )
+
+    note = serializers.CharField(
+        label=_('Notes'),
+        help_text=_('Add transaction note (optional)'),
+        required=False, allow_blank=True,
+    )
+
+    @transaction.atomic
+    def save(self):
+        """Save the serializer to change the status of the selected stock items"""
+
+        data = self.validated_data
+
+        items = data['items']
+        status = data['status']
+
+        request = self.context['request']
+        user = getattr(request, 'user', None)
+
+        note = data.get('note', '')
+
+        items_to_update = []
+        transaction_notes = []
+
+        deltas = {
+            'status': status,
+        }
+
+        now = datetime.now()
+
+        # Instead of performing database updates for each item,
+        # perform bulk database updates (much more efficient)
+
+        for item in items:
+            # Ignore items which are already in the desired status
+            if item.status == status:
+                continue
+
+            item.updated = now
+            item.status = status
+            items_to_update.append(item)
+
+            # Create a new transaction note for each item
+            transaction_notes.append(
+                StockItemTracking(
+                    item=item,
+                    tracking_type=InvenTree.status_codes.StockHistoryCode.EDITED,
```

Comment:
Have you tested this? I think this needs to be

```suggestion
                    tracking_type=InvenTree.status_codes.StockHistoryCode.EDITED.value,
```